CMakeLists.txt

```diff
@@ -502,7 +502,51 @@ add_library(torrent-rasterbar
 	${libtorrent_aux_include_files}
 )
 
-select_cxx_standard(torrent-rasterbar 11)
+# Full C++ 11 support is required
+target_compile_features(torrent-rasterbar
+	PUBLIC
+		cxx_std_11
```

Comment:
it seems risky to declare all of C++11 must be supported. If there's a pedantic interpretation of this, there are compilers that don't support the garbage collection support added to the standard library in C++11 (and nobody uses that anyway).

Reply:
@arvidn 

Well, then, other than `cxx_std_11`, do you think any of these is relevant enough on their own to be here? Like I said, just `cxx_std_11` will also do fine.

Reply:
I'm not sure I know what you mean. I think all of the features below are used (with a few exceptions I pointed out)

Reply:
@arvidn so, have you decided on whether requiring only `cxx_std_11` is sufficient?

Reply:
it would seem to me it would be safer to leave the other, individual, requirements instead. is `cxx_std_11` not an alias for all of the other ones?

Reply:
@arvidn 

> it would seem to me it would be safer to leave the other, individual, requirements instead.

I took it from the previous comments/reviews you would probably want to trim the list a little more still, but if that's not the case, then indeed it is safer this way. I will squash the fixup and mark the PR as ready for review.

> is `cxx_std_11` not an alias for all of the other ones?

Unfortunately, no. In practice it just means "_some_ kind of basic C++11 support".

Reply:
all of the remaining features are used in libtorrent right now, so I don't think it should be trimmed any more.

Reply:
I was under the impression that you would remove the `cxx_std_11` entry though. Did I misunderstand that?

Reply:
@arvidn 

I think it's best to leave it in. Aside from "potential bad side effects" that specifying the individual features but not the high-level meta requirements might have (don't know if any, but best not to risk it), there is this quote from Professional CMake 7th edition:

> In the early days of C++11 and C++14, working with compile features was potentially useful because
compiler support was often lagging behind. For later language releases, support by the mainstream
compilers has arrived much faster. Subsequently, CMake stopped providing fine-grained features
for language standards beyond C++14. For C++17 and later, only the high level meta features like `cxx_std_17` are provided.

So leaving the high level meta feature is more consistent with the future releases; it's only the individually specified features that will get removed.